src/app/util/ClientMonitoringRegistrationTable.cpp

```diff
@@ -66,4 +66,14 @@ ClientMonitoringRegistrationTable::ClientRegistrationEntry & ClientMonitoringReg
     return mRegisteredClient;
 }
 
+CHIP_ERROR ClientMonitoringRegistrationTable::DeleteFromStorage(FabricIndex fabric)
+{
+    return mStorage.SyncDeleteKeyValue(DefaultStorageKeyAllocator::ClientMonitoringTableEntry(fabric).KeyName());
+}
+
+bool ClientMonitoringRegistrationTable::HasValueForFabric(FabricIndex fabric)
+{
+    return mStorage.SyncDoesKeyExist(DefaultStorageKeyAllocator::ClientMonitoringTableEntry(fabric).KeyName());
+}
```

Comment:
This "costs" a lot of keys. It would have been better to store the entire table into one single storage key. This can work for now, but may cause issues if the storage format is improved to reduce number of keys used.

Reply:
Created an issue to revisit this.
#24288 24288